test/crd_validation_test.go

```diff
@@ -0,0 +1,77 @@
+package test
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/validation"
+	"github.com/ghodss/yaml"
+	qdrouterdv1alpha1 "github.com/interconnectedcloud/qdrouterd-operator/pkg/apis/interconnectedcloud/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+var crNameMap = map[string]string{
+	"interconnectedcloud_v1alpha1_qdrouterd_cr.yaml": "interconnectedcloud_v1alpha1_qdrouterd_crd.yaml",
+}
+
+var crdTypeMap = map[string]interface{}{
+	"interconnectedcloud_v1alpha1_qdrouterd_crd.yaml": &qdrouterdv1alpha1.Qdrouterd{},
+}
+
+func TestCRDSchemas(t *testing.T) {
+	for crdFileName, amqType := range crdTypeMap {
+		schema := getSchema(t, crdFileName)
+		missingEntries := schema.GetMissingEntries(amqType)
+		for _, missing := range missingEntries {
+			if strings.HasPrefix(missing.Path, "/status") {
+				//Not using subresources, so status is not expected to appear in CRD
+			} else {
+				assert.Fail(t, "Discrepancy between CRD and Struct",
+					"Missing or incorrect schema validation at %v, expected type %v  in CRD file %v", missing.Path, missing.Type, crdFileName)
+			}
+		}
+	}
+}
+
+func TestSampleCustomResources(t *testing.T) {
+
+	// fetch list of all CRD files
+	fileList, err := ioutil.ReadDir("../deploy/crds")
+	assert.NoError(t, err, "Error reading CR Files %v", fileList)
+
+	var input map[string]interface{}
+	for _, file := range fileList {
+
+		if !strings.HasSuffix(file.Name(), "cr.yaml") {
```

Comment:
These next 10 lines don't make sense for the current context, where there is a single CR and a single CRD. I would either hardcode, or maybe better yet, just iterate over the cr/crd map and get both from there.